x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_data_client.ts

```diff
@@ -34,6 +36,12 @@ type BulkUpsertFromStreamOptions = {
   recordsStream: NodeJS.ReadableStream;
 } & Pick<Parameters<ElasticsearchClient['helpers']['bulk']>[0], 'flushBytes' | 'retries'>;
 
+type StoredAssetCriticalityRecord = {
+  [K in keyof AssetCriticalityRecord]: K extends 'criticality_level'
```

Comment:
![Math_Lady_meme](https://github.com/user-attachments/assets/4e32b03d-99b3-48c1-b34f-5157fdd5b98c)

Reply:
I got it!

Reply:
yeah, idk, any suggestions to improve readability here? 😅 
TS syntax is quite verbose for type computations, but maybe we can extract out the behaviour of "setting the type of a value for a specific key K"?